### ISSUE TITLE
Checking if reservation exists doesnt work.

### ISSUE THREAD

**Issue by zrygan** [zrygan/Hotel-Reservation-System#8]
Test (bold are code input/testcase):
```
1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: **4**
Welcome to the Hotel Reservation System, User!
Please enter your name: **Zry**
Enter the name of the hotel: **max**
Enter the day of your check-in: **1**
Enter the day of your check-out: **2**
Room booked successfully for Zry

1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: **4**
Welcome to the Hotel Reservation System, User!
Please enter your name: **Migo**
Enter the name of the hotel: **max**
Enter the day of your check-in: **1**
Enter the day of your check-out: **2**
Room booked successfully for Migo
```

**Comment by jazjimenez** [comment]
Before that test, did you create a hotel then a room? Cuz if not, then we have to create a function that won't let them go through the simulation if we there aren't any hotels and room yet

**Comment by jazjimenez** [comment]
oh wait nvm, max has 3 rooms na noh

**Comment by jazjimenez** [comment]
the problem most likely relies on the isRoomAvailable in Hotel()

**Comment by jazjimenez** [comment]
hi update! // revisions are in debugging2 branch

``
1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 1 
Enter the name of the hotel: zen 

1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 3
Enter the name of the hotel: zen
1       :        to change the name of the hotel
2       :        to add a new room
3       :        remove a room
4       :        update the base price
5       :        remove a reservation
6       :        remove a hotel
Choice: 2
A new room 'zen_Room_0' has been added in hotel 'zen'.

1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 4
Welcome to the Hotel Reservation System, User!
Please enter your name: jaz
Enter the name of the hotel: zen
Enter the day of your check-in: 1
Enter the day of your check-out: 1
Room booked successfully for jaz
1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 4
Welcome to the Hotel Reservation System, User!
Please enter your name: jaz
Enter the name of the hotel: zen
Enter the day of your check-in: 1
Enter the day of your check-out: 1
There are currently no available rooms for the selected dates
``

so basically i have fixed the problem of the program possibly checking in the same dates.. since we have applied the system of "customer 1 books room_0, custom 2 books _room1, ... and so on". this is fine naman within the specs that quotes

> This mechanism may be automated (e.g. select the first room, if available, return room, if not, check the next room) or manual (e.g. user selects from a list of rooms).

but unfortunately, this system "limits" our program.. cuz in a way, only 50 customers maximum can place a reservation in the hotel... so we aren't capable of "customer 1 books room_0 in day 1 - 10, while customer 2 book room_0 from day 11 - 20"

**Comment by jazjimenez** [comment]
hi update! // revisions are in debugging2 branch

``
1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 1 
Enter the name of the hotel: zen 

1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 3
Enter the name of the hotel: zen
1       :        to change the name of the hotel
2       :        to add a new room
3       :        remove a room
4       :        update the base price
5       :        remove a reservation
6       :        remove a hotel
Choice: 2
A new room 'zen_Room_0' has been added in hotel 'zen'.

1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 4
Welcome to the Hotel Reservation System, User!
Please enter your name: jaz
Enter the name of the hotel: zen
Enter the day of your check-in: 1
Enter the day of your check-out: 1
Room booked successfully for jaz
1       :        Create Hotel
2       :        View Hotel
3       :        Manage Hotel
4       :        Simulate Booking
0       :        Exit
9
Choose an option: 4
Welcome to the Hotel Reservation System, User!
Please enter your name: jaz
Enter the name of the hotel: zen
Enter the day of your check-in: 1
Enter the day of your check-out: 1
There are currently no available rooms for the selected dates
``

so basically i have fixed the problem of the program possibly checking in the same dates.. since we have applied the system of "customer 1 books room_0, custom 2 books _room1, ... and so on". this is fine naman within the specs that quotes

> This mechanism may be automated (e.g. select the first room, if available, return room, if not, check the next room) or manual (e.g. user selects from a list of rooms).

but unfortunately, this system "limits" our program.. cuz in a way, only 50 customers maximum can place a reservation in the hotel... so we aren't capable of "customer 1 books room_0 in day 1 - 10, while customer 2 book room_0 from day 11 - 20"

**Comment by jazjimenez** [comment]
lastly, the next problem is in the view hotel.. since its not seeing that "room_0" is booked...

**Comment by zrygan** [comment]
See commit `Fixed Issue #8`.
